cmd/hostd/run.go

```diff
@@ -41,6 +42,32 @@ import (
 	"lukechampine.com/upnp"
 )
 
+func defaultDataDirectory(fp string) string {
+	// use the provided path if it's not empty
+	if fp != "" {
+		return fp
+	}
+
+	// check for databases in the current directory
+	if _, err := os.Stat("hostd.db"); err == nil {
+		return ""
+	} else if _, err := os.Stat("hostd.sqlite3"); err == nil {
+		return ""
+	}
+
+	// default to the operating system's application directory
+	switch runtime.GOOS {
```

Comment:
I'd consider moving this whole switch statement into a helper method to avoid duplicating it 3 times. For the one case where we want to be compatible with the old location we can add a dedicated
```
switch runtime.GOOS {
  case "linux", "freebsd", "openbsd":
    // add old default
}
```
and still save a bunch of lines

Reply:
The three are subtly different:

1. `run.go:59` returns `/var/lib/hostd` on Linux
2. `config.go:258` returns `/etc/hostd/hostd.yml` on Linux
3. `main.go:110`, as you said, returns multiple paths